### ISSUE TITLE
[BUG] DataFrame iloc[:0] does not work after python refactor

### ISSUE THREAD

**Issue by dantegd** [rapidsai/cudf#3715]
**Describe the bug**
Before the refactor this would work perfectly, where `df` is a cuDF dataframe: 

```python
>> df.iloc[:0]
Empty DataFrame
Columns: [a, b]
Index: []
```

After the refactor the following error happens:

```python
>> df.iloc[:0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dante/miniconda3/envs/ns0107/lib/python3.7/site-packages/cudf/core/dataframe.py", line 748, in __repr__
    output = self.get_renderable_dataframe()
  File "/home/dante/miniconda3/envs/ns0107/lib/python3.7/site-packages/cudf/core/dataframe.py", line 734, in get_renderable_dataframe
    self._data[col].has_nulls
  File "cudf/_libxx/column.pyx", line 110, in cudf._libxx.column.Column.has_nulls.__get__
  File "/home/dante/miniconda3/envs/ns0107/lib/python3.7/site-packages/cudf/utils/utils.py", line 323, in __get__
    value = self.func(instance)
  File "cudf/_libxx/column.pyx", line 130, in cudf._libxx.column.Column.null_count
  File "cudf/_libxx/column.pyx", line 133, in cudf._libxx.column.Column.compute_null_count
  File "cudf/_libxx/column.pyx", line 196, in cudf._libxx.column.Column._view
TypeError: an integer is required
```

**Expected behavior**
No error, getting the expected output

**Environment overview (please complete the following information)**
 - Environment location: Bare metal
 - Method of cuDF install: conda nightly 

**Environment details**
<details><summary>Click here to see environment details</summary><pre>

     **git***
     commit 26ade6d257698a25a22a8d349e62f3be5d196f56 (HEAD -> branch-0.12, origin/branch-0.12, origin/HEAD)
     Merge: 4a18029c6 201e9eba2
     Author: Keith Kraus <kkraus@nvidia.com>
     Date:   Tue Jan 7 17:11:30 2020 -0500

     Merge pull request #3254 from shwina/libcudfxx-python

     [REVIEW] Python redesign for libcudf++
     **git submodules***
     b165e1fb11eeea64ccf95053e40f2424312599cc thirdparty/cub (v1.7.1)
     bcd545071c7a5ddb28cb6576afc6399eb1286c43 thirdparty/jitify (heads/cudf)
     cdcda484d0c7db114ea29c3b33429de5756ecfd8 thirdparty/libcudacxx (0.8.1-99-gcdcda48)
     a97a7380c76346c22bb67b93695bed19592afad2 thirdparty/libcudacxx/libcxx (heads/rapidsai-interop)

     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=18.04
     DISTRIB_CODENAME=bionic
     DISTRIB_DESCRIPTION="Ubuntu 18.04.3 LTS"
     NAME="Ubuntu"
     VERSION="18.04.3 LTS (Bionic Beaver)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 18.04.3 LTS"
     VERSION_ID="18.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=bionic
     UBUNTU_CODENAME=bionic
     Linux gv100-1804 5.0.0-37-generic #40~18.04.1-Ubuntu SMP Thu Nov 14 12:06:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

     ***GPU Information***
     Wed Jan  8 02:11:08 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 430.40       Driver Version: 430.40       CUDA Version: 10.1     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Quadro GV100        Off  | 00000000:17:00.0 Off |                  Off |
     | 30%   42C    P2    26W / 250W |   3979MiB / 32508MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   1  Quadro GV100        Off  | 00000000:65:00.0  On |                  Off |
     | 30%   43C    P0    28W / 250W |    534MiB / 32505MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+

     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |    0     10533      C   ...dante/miniconda3/envs/ns0107/bin/python  3968MiB |
     |    1      1555      G   /usr/lib/xorg/Xorg                            71MiB |
     |    1      2070      G   /usr/bin/gnome-shell                          62MiB |
     |    1      3248      G   /usr/lib/xorg/Xorg                           197MiB |
     |    1      3401      G   /usr/bin/gnome-shell                         183MiB |
     |    1      4412      G   gnome-control-center                           7MiB |
     +-----------------------------------------------------------------------------+

     ***CPU***
     Architecture:        x86_64
     CPU op-mode(s):      32-bit, 64-bit
     Byte Order:          Little Endian
     CPU(s):              12
     On-line CPU(s) list: 0-11
     Thread(s) per core:  2
     Core(s) per socket:  6
     Socket(s):           1
     NUMA node(s):        1
     Vendor ID:           GenuineIntel
     CPU family:          6
     Model:               85
     Model name:          Intel(R) Core(TM) i7-7800X CPU @ 3.50GHz
     Stepping:            4
     CPU MHz:             1200.073
     CPU max MHz:         4000.0000
     CPU min MHz:         1200.0000
     BogoMIPS:            7000.00
     Virtualization:      VT-x
     L1d cache:           32K
     L1i cache:           32K
     L2 cache:            1024K
     L3 cache:            8448K
     NUMA node0 CPU(s):   0-11
     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req md_clear flush_l1d

     ***CMake***
     /home/dante/miniconda3/envs/ns0107/bin/cmake
     cmake version 3.14.5

     CMake suite maintained and supported by Kitware (kitware.com/cmake).

     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
     Copyright (C) 2017 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

     ***nvcc***
     /usr/local/cuda-10.1u2/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2019 NVIDIA Corporation
     Built on Sun_Jul_28_19:07:16_PDT_2019
     Cuda compilation tools, release 10.1, V10.1.243

     ***Python***
     /home/dante/miniconda3/envs/ns0107/bin/python
     Python 3.7.6

     ***Environment Variables***
     PATH                            : /usr/local/cuda-10.1u2/bin:/home/dante/miniconda3/envs/ns0107/bin:/home/dante/miniconda3/condabin:/usr/local/cuda-10.1u2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
     LD_LIBRARY_PATH                 : /usr/local/cuda/lib64
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/dante/miniconda3/envs/ns0107
     PYTHON_PATH                     :

     ***conda packages***
     /home/dante/miniconda3/condabin/conda
     # packages in environment at /home/dante/miniconda3/envs/ns0107:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     arrow-cpp                 0.15.0           py37h090bef1_2    conda-forge
     attrs                     19.3.0                     py_0    conda-forge
     bokeh                     1.4.0                    py37_0    conda-forge
     boost-cpp                 1.70.0               h8e57a91_2    conda-forge
     brotli                    1.0.7             he1b5a44_1000    conda-forge
     bzip2                     1.0.8                h516909a_2    conda-forge
     c-ares                    1.15.0            h516909a_1001    conda-forge
     ca-certificates           2019.11.28           hecc5488_0    conda-forge
     certifi                   2019.11.28               py37_0    conda-forge
     click                     7.0                        py_0    conda-forge
     cloudpickle               1.2.2                      py_1    conda-forge
     cmake                     3.14.5               hf94ab9c_0    conda-forge
     cudatoolkit               10.1.243             h6bb024c_0    nvidia
     cudf                      0.12.0b200107          py37_835    rapidsai-nightly
     cudnn                     7.6.0                cuda10.1_0    nvidia
     cuml                      0.12.0a0+417.g0563c27c          pypi_0    pypi
     cupy                      6.6.0            py37ha7c4746_1    conda-forge
     curl                      7.65.3               hf8cf82a_0    conda-forge
     cython                    0.29.14          py37he1b5a44_0    conda-forge
     cytoolz                   0.10.1           py37h516909a_0    conda-forge
     dask                      2.9.1+5.g1d86a12a          pypi_0    pypi
     dask-cuda                 0.12.0a200107           py37_47    rapidsai-nightly
     dask-cudf                 0.12.0b200107          py37_835    rapidsai-nightly
     distributed               2.9.1+6.gf68119a4          pypi_0    pypi
     dlpack                    0.2                  he1b5a44_1    conda-forge
     double-conversion         3.1.5                he1b5a44_2    conda-forge
     expat                     2.2.5             he1b5a44_1004    conda-forge
     fastavro                  0.22.9           py37h516909a_0    conda-forge
     fastrlock                 0.4             py37he1b5a44_1000    conda-forge
     freetype                  2.10.0               he983fc9_1    conda-forge
     fsspec                    0.6.2                      py_0    conda-forge
     gflags                    2.2.2             he1b5a44_1002    conda-forge
     glog                      0.4.0                he1b5a44_1    conda-forge
     grpc-cpp                  1.23.0               h18db393_0    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     icu                       64.2                 he1b5a44_1    conda-forge
     importlib_metadata        1.3.0                    py37_0    conda-forge
     jinja2                    2.10.3                     py_0    conda-forge
     joblib                    0.14.1                     py_0    conda-forge
     jpeg                      9c                h14c3975_1001    conda-forge
     krb5                      1.16.4               h2fd8d38_0    conda-forge
     ld_impl_linux-64          2.33.1               h53a641e_7    conda-forge
     libblas                   3.8.0               14_openblas    conda-forge
     libcblas                  3.8.0               14_openblas    conda-forge
     libclang                  8.0.0                h6bb024c_0    rapidsai
     libcudf                   0.12.0b200107      cuda10.1_835    rapidsai-nightly
     libcumlprims              0.12.0a200107        cuda10.1_0    rapidsai-nightly
     libcurl                   7.65.3               hda55be3_0    conda-forge
     libedit                   3.1.20170329      hf8c457e_1001    conda-forge
     libevent                  2.1.10               h72c5cf5_0    conda-forge
     libffi                    3.2.1             he1b5a44_1006    conda-forge
     libgcc-ng                 9.2.0                h24d8f2e_1    conda-forge
     libgfortran-ng            7.3.0                hdf63c60_3    conda-forge
     libgomp                   9.2.0                h24d8f2e_1    conda-forge
     libhwloc                  2.1.0                h3c4fd83_0    conda-forge
     libiconv                  1.15              h516909a_1005    conda-forge
     liblapack                 3.8.0               14_openblas    conda-forge
     libnvstrings              0.12.0b200107      cuda10.1_835    rapidsai-nightly
     libopenblas               0.3.7                h5ec1e0e_6    conda-forge
     libpng                    1.6.37               hed695b0_0    conda-forge
     libprotobuf               3.8.0                h8b12597_0    conda-forge
     librmm                    0.12.0a200108      cuda10.1_153    rapidsai-nightly
     libssh2                   1.8.2                h22169c7_2    conda-forge
     libstdcxx-ng              9.2.0                hdf63c60_1    conda-forge
     libtiff                   4.1.0                hfc65ed5_0    conda-forge
     libuv                     1.34.0               h516909a_0    conda-forge
     libxml2                   2.9.10               hee79883_0    conda-forge
     llvmlite                  0.29.0           py37hfd453ef_1    conda-forge
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.8.3             he1b5a44_1001    conda-forge
     markupsafe                1.1.1            py37h516909a_0    conda-forge
     more-itertools            8.0.2                      py_0    conda-forge
     msgpack-python            0.6.2            py37hc9558a2_0    conda-forge
     nccl                      2.4.6.1              cuda10.1_0    nvidia
     ncurses                   6.1               hf484d3e_1002    conda-forge
     numba                     0.45.1           py37hb3f55d8_0    conda-forge
     numpy                     1.17.3           py37h95a1406_0    conda-forge
     nvstrings                 0.12.0b200107          py37_835    rapidsai-nightly
     olefile                   0.46                       py_0    conda-forge
     openmp_impl               4.5                       0_gnu    conda-forge
     openssl                   1.1.1d               h516909a_0    conda-forge
     packaging                 20.0                       py_0    conda-forge
     pandas                    0.24.2           py37hb3f55d8_1    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     partd                     1.1.0                      py_0    conda-forge
     patsy                     0.5.1                      py_0    conda-forge
     pillow                    7.0.0            py37hefe7db6_0    conda-forge
     pip                       19.3.1                   py37_0    conda-forge
     pluggy                    0.13.0                   py37_0    conda-forge
     protobuf                  3.8.0            py37he1b5a44_2    conda-forge
     psutil                    5.6.7            py37h516909a_0    conda-forge
     py                        1.8.1                      py_0    conda-forge
     pyarrow                   0.15.0           py37h8b68381_1    conda-forge
     pynvml                    8.0.3                      py_0    conda-forge
     pyparsing                 2.4.6                      py_0    conda-forge
     pytest                    5.3.2                    py37_0    conda-forge
     python                    3.7.6                h357f687_2    conda-forge
     python-dateutil           2.8.1                      py_0    conda-forge
     pytz                      2019.3                     py_0    conda-forge
     pyyaml                    5.3              py37h516909a_0    conda-forge
     re2                       2020.01.01           he1b5a44_0    conda-forge
     readline                  8.0                  hf8c457e_0    conda-forge
     rhash                     1.3.6             h14c3975_1001    conda-forge
     rmm                       0.12.0a200108          py37_153    rapidsai-nightly
     scikit-learn              0.22.1           py37hcdab131_0    conda-forge
     scipy                     1.4.1            py37h921218d_0    conda-forge
     setuptools                44.0.0                   py37_0    conda-forge
     six                       1.13.0                   py37_0    conda-forge
     snappy                    1.1.7             he1b5a44_1003    conda-forge
     sortedcontainers          2.1.0                      py_0    conda-forge
     sqlite                    3.30.1               hcee41ef_0    conda-forge
     statsmodels               0.10.2           py37hc1659b7_0    conda-forge
     tblib                     1.6.0                      py_0    conda-forge
     thrift-cpp                0.12.0            hf3afdfd_1004    conda-forge
     tk                        8.6.10               hed695b0_0    conda-forge
     toolz                     0.10.0                     py_0    conda-forge
     tornado                   6.0.3            py37h516909a_0    conda-forge
     ucx                       1.7.0dev+g430ae7e      cuda10.1_0    rapidsai
     ucx-py                    0.12.0a200107+g553a09e         py37_83    rapidsai-nightly
     umap-learn                0.3.10                   py37_1    conda-forge
     uriparser                 0.9.3                he1b5a44_1    conda-forge
     wcwidth                   0.1.8                      py_0    conda-forge
     wheel                     0.33.6                   py37_0    conda-forge
     xz                        5.2.4             h14c3975_1001    conda-forge
     yaml                      0.2.2                h516909a_1    conda-forge
     zict                      1.0.0                      py_0    conda-forge
     zipp                      0.6.0                      py_0    conda-forge
     zlib                      1.2.11            h516909a_1006    conda-forge
     zstd                      1.4.3                h3b9ef0a_0    conda-forge

</pre></details>

**Additional context**
This is causing an issue is cuml.dask that is blocking CI, relevant code: https://github.com/rapidsai/cuml/blob/0563c27c4fe768ff0a4e1a60c2961fdfa9377ef2/python/cuml/dask/common/dask_df_utils.py#L58

Might be able to implement a workaround tomorrow, but this bug should be solved in cuDF to match the Pandas behavior

**Comment by dantegd** [comment]
Update: I have been able to reproduce this locally in my workstation (bare metal) using all the conda nightlies released since the libcudf++ Python refactor was merged. But if I build cuDF from source then everything works as expected, and it seems to be working in CI too, so there's a high chance this might be something environment related, will update with further findings.

**Comment by dantegd** [comment]
@kkraus14 @shwina narrowed this issue to environments with numba 0.45, updating to 0.46 fixes the issue in all my environments

**Comment by kkraus14** [comment]
Fixed by #3734
